### DISCUSSION TITLE
服务器部署服务端配置文件报错

### DISCUSSION THREAD

**Discussion by NZESupB** [zdz/ServerStatus-Rust#30] [Q&A] [answered]
用默认的配置文件啥也不改也是提示这个.
默认配置下检测与我修改后检测都是提示这个.1.4的release.
./stat_server -c config.toml
✨ run in normal mode, load conf from local file `config.toml
✨ admin_user: admin
✨ admin_pass: cafa1cfa-2fb6-4ae4-9df6-9cfe70b54443
Error: EOF while parsing a value at line 1 column 0

**Comment by zdz** [comment]
<img width="735" alt="image" src="https://user-images.githubusercontent.com/152173/166097490-5573c415-4c3a-4b5b-8691-59d50f76fce0.png">

亲测没问题，估计是改的格式不对

**Comment by zdz** [comment]
对对格式，或者用 这个 ```RUST_BACKTRACE=full RUST_LOG=trace ./stat_server -c config.toml``` 打出详细日志

**Comment by NZESupB** [comment]
我这边用默认配置是打印这个.

**Comment by zdz** [comment]
额，输出或代码前后用这个 ``` 包一下呢，没法看

**Comment by NZESupB** [comment]
``` ✨ run in normal mode, load conf from local file `config.toml
✨ admin_user: admin
✨ admin_pass: 7eaaae57-a374-436a-9224-e00cb48fa533
 DEBUG stat_server > Config { http_addr: "0.0.0.0:8080", grpc_addr: "0.0.0.0:9394", notify_interval: 30, offline_threshold: 30, admin_user: Some("admin"), admin_pass: Some("7eaaae57-a374-436a-9224-e00cb48fa533"), tgbot: Config { enabled: false, bot_token: "<tg bot token>", chat_id: "<chat id>", title: "❗<b>Server Status</b>", online_tpl: "{{config.title}} \n😆 {{host.location}} {{host.name}} 主机恢复上线啦", offline_tpl: "{{config.title}} \n😱 {{host.location}} {{host.name}} 主机已经掉线啦", custom_tpl: "{% if host.memory_used / host.memory_total > 0.5  %}\n<pre>😲 {{host.name}} 主机内存使用率超50%, 当前{{ (100 * host.memory_used / host.memory_total) | round }}%  </pre>\n{% endif %}\n\n{% if host.hdd_used / host.hdd_total  > 0.5  %}\n<pre>😲 {{host.name}} 主机硬盘使用率超50%, 当前{{ (100 * host.hdd_used / host.hdd_total) | round }}% </pre>\n{% endif %}\n" }, wechat: Config { enabled: false, corp_id: "<corp id>", corp_secret: "<corp secret>", agent_id: "<agent id>", title: "❗Server Status", online_tpl: "{{config.title}} \n😆 {{host.location}} 的 {{host.name}} 主机恢复上线啦", offline_tpl: "{{config.title}} \n😱 {{host.location}} 的 {{host.name}} 主机已经掉线啦", custom_tpl: "{% if host.memory_used / host.memory_total > 0.8  %}\n😲 {{host.name}} 主机内存使用率超80%\n{% endif %}\n\n{% if host.hdd_used / host.hdd_total > 0.8  %}\n😲 {{host.name}} 主机硬盘使用率超80%\n{% endif %}\n" }, email: Config { enabled: false, server: "smtp.gmail.com", username: "user@email.com", password: "***", to: "user1@email.com;user2@email.com", subject: "ServerStatus Notification", title: "❗<b>Server Status</b><br/>", online_tpl: "{{config.title}} 😆 {{host.location}} 的 {{host.name}} 主机恢复上线啦", offline_tpl: "{{config.title}} 😱 {{host.location}} 的 {{host.name}} 主机已经掉线啦", custom_tpl: "{% if host.memory_used / host.memory_total > 0.8  %}\n<pre>😲 {{host.name}} 主机内存使用率超80%, 当前{{ (100 * host.memory_used / host.memory_total) | round }}%  </pre>\n{% endif %}\n\n{% if host.hdd_used / host.hdd_total > 0.8  %}\n<pre>😲 {{host.name}} 主机硬盘使用率超80%, 当前{{ (100 * host.hdd_used / host.hdd_total) | round }}% </pre>\n{% endif %}\n" }, hosts: [Host { name: "h1", password: "p1", alias: "n1", location: "🏠", host_type: "kvm", monthstart: 1, notify: true, disabled: false, last_network_in: 0, last_network_out: 0, pos: 0 }, Host { name: "h2", password: "p2", alias: "n2", location: "🏢", host_type: "kvm", monthstart: 1, notify: true, disabled: false, last_network_in: 0, last_network_out: 0, pos: 1 }, Host { name: "h3", password: "p3", alias: "n3", location: "🏝\u{fe0f}", host_type: "kvm", monthstart: 1, notify: true, disabled: false, last_network_in: 0, last_network_out: 0, pos: 2 }], hosts_map: {"h2": Host { name: "h2", password: "p2", alias: "n2", location: "🏢", host_type: "kvm", monthstart: 1, notify: true, disabled: false, last_network_in: 0, last_network_out: 0, pos: 1 }, "h3": Host { name: "h3", password: "p3", alias: "n3", location: "🏝\u{fe0f}", host_type: "kvm", monthstart: 1, notify: true, disabled: false, last_network_in: 0, last_network_out: 0, pos: 2 }, "h1": Host { name: "h1", password: "p1", alias: "n1", location: "🏠", host_type: "kvm", monthstart: 1, notify: true, disabled: false, last_network_in: 0, last_network_out: 0, pos: 0 }} }```
Error: EOF while parsing a value at line 1 column 0

**Reply by NZESupB** [reply]
```bash
DEBUG stat_server > 
Config { 
http_addr: "0.0.0.0:8080", 
grpc_addr: "0.0.0.0:9394", 
notify_interval: 30, 
offline_threshold: 30, 
admin_user: Some("admin"), 
admin_pass: Some("7eaaae57-a374-436a-9224-e00cb48fa533"), 
tgbot: Config 
{
enabled: false, 
bot_token: "<tg bot token>", 
chat_id: "<chat id>", 
title: "❗<b>Server Status</b>", 
online_tpl: "{{config.title}} \n😆 {{host.location}} {{host.name}} 主机恢复上线啦", 
offline_tpl: "{{config.title}} \n😱 {{host.location}} {{host.name}} 主机已经掉线啦", 
custom_tpl: "{% if host.memory_used / host.memory_total > 0.5  %}\n<pre>😲 {{host.name}} 主机内存使用率超50%, 当前{{ (100 * host.memory_used / host.memory_total) | round }}%  </pre>\n{% endif %}\n\n{% if host.hdd_used / host.hdd_total  > 0.5  %}\n<pre>😲 {{host.name}} 主机硬盘使用率超50%, 当前{{ (100 * host.hdd_used / host.hdd_total) | round }}% </pre>\n{% endif %}\n" 
}, 

wechat: Config 
{ 
enabled: false, 
corp_id: "<corp id>", 
corp_secret: "<corp secret>", 
agent_id: "<agent id>", 
title: "❗Server Status", 
online_tpl: "{{config.title}} \n😆 {{host.location}} 的 {{host.name}} 主机恢复上线啦", 
offline_tpl: "{{config.title}} \n😱 {{host.location}} 的 {{host.name}} 主机已经掉线啦", 
custom_tpl: "{% if host.memory_used / host.memory_total > 0.8  %}\n😲 {{host.name}} 主机内存使用率超80%\n{% endif %}\n\n{% if host.hdd_used / host.hdd_total > 0.8  %}\n😲 {{host.name}} 主机硬盘使用率超80%\n{% endif %}\n" 
}, 

email: Config 
{ 
enabled: false, 
server: "smtp.gmail.com", 
username: "user@email.com", 
password: "***", 
to: "user1@email.com;user2@email.com", 
subject: "ServerStatus Notification", 
title: "❗<b>Server Status</b><br/>", 
online_tpl: "{{config.title}} 😆 {{host.location}} 的 {{host.name}} 主机恢复上线啦", 
offline_tpl: "{{config.title}} 😱 {{host.location}} 的 {{host.name}} 主机已经掉线啦", 
custom_tpl: "{% if host.memory_used / host.memory_total > 0.8  %}\n<pre>😲 {{host.name}} 主机内存使用率超80%, 当前{{ (100 * host.memory_used / host.memory_total) | round }}%  </pre>\n{% endif %}\n\n{% if host.hdd_used / host.hdd_total > 0.8  %}\n<pre>😲 {{host.name}} 主机硬盘使用率超80%, 当前{{ (100 * host.hdd_used / host.hdd_total) | round }}% </pre>\n{% endif %}\n" 
}, 

hosts: [
Host { name: "h1", password: "p1", alias: "n1", location: "🏠", host_type: "kvm", monthstart: 1, notify: true, disabled: false, last_network_in: 0, last_network_out: 0, pos: 0 }, 
Host { name: "h2", password: "p2", alias: "n2", location: "🏢", host_type: "kvm", monthstart: 1, notify: true, disabled: false, last_network_in: 0, last_network_out: 0, pos: 1 }, 
Host { name: "h3", password: "p3", alias: "n3", location: "🏝\u{fe0f}", host_type: "kvm", monthstart: 1, notify: true, disabled: false, last_network_in: 0, last_network_out: 0, pos: 2 }
], 

hosts_map: 
{
"h2": Host { name: "h2", password: "p2", alias: "n2", location: "🏢", host_type: "kvm", monthstart: 1, notify: true, disabled: false, last_network_in: 0, last_network_out: 0, pos: 1 }, 
"h3": Host { name: "h3", password: "p3", alias: "n3", location: "🏝\u{fe0f}", host_type: "kvm", monthstart: 1, notify: true, disabled: false, last_network_in: 0, last_network_out: 0, pos: 2 }, 
"h1": Host { name: "h1", password: "p1", alias: "n1", location: "🏠", host_type: "kvm", monthstart: 1, notify: true, disabled: false, last_network_in: 0, last_network_out: 0, pos: 0 }
} 
}
```

**Reply by NZESupB** [reply]
分割了一下

**Comment by zdz** [comment]
额。看配置是没啥问题

**Reply by NZESupB** [reply]
所以就是不知道是哪出问题了..这个就是直接wget的默认配置..client的话我还没配置.是要client启动了才行么.

**Comment by zdz** [comment]
zip 包里面有

**Comment by zdz** [comment]
<img width="1471" alt="image" src="https://user-images.githubusercontent.com/152173/166098740-d9a55372-18b6-426f-8935-863d43473697.png">

wget 的是 raw ?

**Reply by NZESupB** [reply]
对,就是wget --no-check-certificate -qO config.toml 'https://raw.githubusercontent.com/zdz/ServerStatus-Rust/master/config.toml'

**Reply by NZESupB** [reply]
后面我用release的压缩包解压估计也覆盖掉了

**Comment by zdz** [comment]
<img width="1499" alt="image" src="https://user-images.githubusercontent.com/152173/166098811-9748ea43-8382-403f-b116-47ebdb1544f7.png">
我试了都ok。。

**Reply by NZESupB** [reply]
所以我也觉得很迷惑..docker也是这样报这个错.难道是系统有啥问题么..

**Comment by zdz** [comment]
什么系统？

**Reply by NZESupB** [reply]
debian11. Linux debian 5.10.0-9-amd64 #1 SMP Debian 5.10.70-1 (2021-09-30) x86_64 GNU/Linux

**Comment by zdz** [comment]
方便的话可以创建个非root账号发我邮箱我瞧瞧

**Reply by NZESupB** [reply]
好,我弄下,大佬你的邮箱是啥

**Comment by zdz** [comment]


**Reply by NZESupB** [reply]
发送了

**Reply by zdz** [reply]
好的

**Comment by zdz** [comment, marked as answer]
<img width="1529" alt="image" src="https://user-images.githubusercontent.com/152173/166099705-9bf7cded-ff8a-4ca4-a3a2-0d70e2820b47.png">

没问题，你切到那个测试账号看看，server 可以不需要root来跑，没问题就把测试账号删了吧

**Reply by NZESupB** [reply]
好奇怪..为啥就root账户不行.把测试号的配置复制过来都这样..玄学

**Reply by zdz** [reply]
root和非root 有的 env 不一样，你可以跑env 命令 diff 一下

**Reply by NZESupB** [reply]
我root账户进测试的文件夹也是可以启动的.难道我下的东西有问题么..

**Reply by zdz** [reply]
md5sum 一下看看指纹

**Reply by NZESupB** [reply]
MD5是一样的..

**Reply by NZESupB** [reply]
是不是因为stats.json这个文件.

**Reply by zdz** [reply]
首次运行是没有这个文件的，不存在解析异常的情况

**Reply by NZESupB** [reply]
哦,那就是这个问题了.我touch了个stats.json就会报这个错.看来docker不能首次运行就映射这个文件.

**Reply by zdz** [reply]
赞一个，我后面兼容一下

**Reply by NZESupB** [reply]
嗯,确实是这样.是初始化的时候就创建了这个空白文件的问题..我把他映射删除之后就全正常了.感谢开发者.
